### ISSUE TITLE
Model collapses after fine-tuning on the toy data

### ISSUE THREAD

**Issue by SamorsetTuska** [ByteDance-Seed/Bagel#292]
**Description**
After fine-tuning Bagel using the vlm subset of the toy data, the model appears to have collapsed. It has lost its basic language understanding capabilities and produces incoherent text, even for samples from the training set or simple prompts.

Here is my related code:
**example_vlm.yaml**
```
vlm_sft:
  dataset_names:
  - llava_ov
  image_transform_args:
    image_stride: 14
    max_image_size: 980
    min_image_size: 378
    max_pixels: 2_007_040
  frame_sampler_args:
    max_num_frames: 12
    min_num_frames: 8
  is_mandatory: true
  shuffle_lines: True
  shuffle_seed: 0
  num_used_data:
  - 1000
  weight: 1
```
**Training Script**
```
# Fine-tuning  
torchrun \
  --nnodes=1 \
  --node_rank=0 \
  --nproc_per_node=8 \
  --master_addr=localhost \
  --master_port=29500 \
  train/pretrain_unified_navit.py \
  --dataset_config_file ./data/configs/example_vlm.yaml \
  --model_path ./models/BAGEL-7B-MoT \
  --layer_module Qwen2MoTDecoderLayer \
  --max_latent_size 64 \
  --finetune_from_hf True \
  --resume-model-only True \
  --finetune-from-ema True \
  --lr 2e-5 \
  --num_worker 4 \
  --expected_num_tokens 16384 \
  --max_num_tokens 36864 \
  --max_num_tokens_per_sample 16384 \
  --wandb_offline True \
  --sharding_strategy FULL_SHARD \
  --freeze_llm False \
  --freeze_vit True \
  --freeze_vae True \
  --freeze_und False \
  --visual_gen False \
  --visual_und True \
  --gradient_accumulation_steps 1 \
  --results_dir ./output/demo_vlm \
  --checkpoint_dir ./output/demo_vlm/checkpoints \
  --total_steps 100 \
  --warmup_steps 10 \
  --log_every 1 \
  --save_every 5 \
  --auto_resume False \
  --resume_from None \
```

**Merging Script**
```
import argparse
import torch.distributed.checkpoint as dist_cp
import torch.distributed.checkpoint.format_utils as dist_cp_format_utils
import torch
from safetensors.torch import save_file, load_file

ckpt_dir = './output/demo_vlm/checkpoints/0000100/model'
save_path = './models/BAGEL-7B-MoT-demo-und/model.safetensors'

state_dict = {}
print(f"Loading checkpoint from {ckpt_dir}...")
dist_cp_format_utils._load_state_dict(
state_dict,
storage_reader=dist_cp.FileSystemReader(ckpt_dir),
planner=dist_cp_format_utils._EmptyStateDictLoadPlanner(),
no_dist=True,
)

if len(state_dict.keys()) == 1:
    print("Unwrapping state_dict...")
    state_dict = state_dict[list(state_dict)[0]]

print("Converting tensors to bfloat16...")
for key, value in state_dict.items():
    if isinstance(value, torch.Tensor):
        if value.is_floating_point():
            state_dict[key] = value.to(dtype=torch.bfloat16)

save_file(state_dict, save_path)
print(f'Successfully merged checkpoint to {save_path}')
```

**Inference Script**
```
#Including 2 samples from the training set and 2 simple conversations.
import sys
import os
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
from PIL import Image, ImageFile
from eval.vlm.utils import load_model_and_tokenizer, build_transform, process_conversation

model_path = "models/BAGEL-7B-MoT-demo-und"

class Args:
    def __init__(self, path):
        self.model_path = path

args = Args(model_path)

model, tokenizer, new_token_ids = load_model_and_tokenizer(args, device="cuda:0", ema=False)  # load model.safetensors
image_transform = build_transform()

data_root_vlm = "data_demo/bagel_example/vlm/images"

sample_1 = {
    "images": ["b73341fba44bee19.jpg"], 
    "conversations": [
        {
            "from": "human", 
            "value": "Create a compact narrative representing the image presented."
            }, 
        {
            "from": "gpt", 
            "value": "A computer is open to a website in which a customization page is open that is titled Themes."
            }
        ]
    }

sample_2 = {
    "images": ["10751.jpeg"], 
    "conversations": [
        {
            "from": "human", 
            "value": "How many days it takes to review the clinical trial applications in Canada?\nAnswer the question using a single word or phrase."
            }, 
        {
            "from": "gpt", 
            "value": "30 Days"
            }
        ]
    }

sample_3 = {
    "images": [], 
    "conversations": [
        {
            "from": "human", 
            "value": "Say 'Hello World'"
            }, 
        {
            "from": "gpt", 
            "value": "Hello World"
            }
        ]
    }

sample_4 = {
    "images": [], 
    "conversations": [
        {
            "from": "human", 
            "value": "1+1="
            }, 
        {
            "from": "gpt", 
            "value": "2"
            }
        ]
    }

def eval(model, sample, data_root):
    if len(sample["images"]) > 0:
        images = [Image.open(os.path.join(data_root, img_path)).convert("RGB") for img_path in sample["images"]]
    else:
        images = []
    question = sample["conversations"][0]["value"]

    images, conversation = process_conversation(images, question)

    response = model.chat(
        tokenizer,
        new_token_ids,
        image_transform,
        images=images,
        prompt=conversation,
        max_length=24,
    )

    print(response.strip().upper())

if __name__ == "__main__":
    print("\nEvaluating sample 1:")
    eval(model, sample_1, data_root_vlm)
    print("\nEvaluating sample 2:")
    eval(model, sample_2, data_root_vlm)
    print("\nEvaluating sample 3:")
    eval(model, sample_3, data_root_vlm)
    print("\nEvaluating sample 4:")
    eval(model, sample_4, data_root_vlm)
```

Here are the inference results:
checkpoints for 100 training steps:
<img width="885" height="226" alt="Image" src="https://github.com/user-attachments/assets/6321ec2b-c2c3-4aaf-9ea8-97b93243ce2d" />
checkpoints for 5 training steps:
<img width="1034" height="204" alt="Image" src="https://github.com/user-attachments/assets/0c9476ed-05aa-48d6-aeae-8258b9c210a5" />
official checkpoints:
<img width="648" height="188" alt="Image" src="https://github.com/user-attachments/assets/128405cc-1fa4-4731-b4d5-d16c68dd1f37" />

This issue occurs with both `model.safetensors` and `ema.safetensors`.

**Comment by SamorsetTuska** [comment]
It's because `--resume_from ` is set `None`
